plugins/output/adapt/helpers.js

```diff
@@ -206,33 +209,69 @@ function checkFrameworkVersion(versionMetaData, cb) {
-      includePrerelease: true
-    })) {
-      cb();
-    } else {
-      cb(new ImportError('Import version (' + importVersion + ') not compatible with installed version (' + installedVersion + ')', 400));
+    if (!importedVersion) {
+      return cb(new ImportError('Invalid version number (' + importedVersion + ') found in import package.json'), 400)
     }
+    cb(null, { installed: installedVersion, imported: importedVersion});
   });
 };
 
-function checkPluginFrameworkVersion(serverFrameworkVersion, pluginMetaData) {
-  const validFrameworkVersion = (semver.valid(pluginMetaData.framework) || semver.validRange(pluginMetaData.framework));
-  if (!validFrameworkVersion) {
-    return new ImportError(`Invalid version number (${pluginMetaData.framework}) found in ${pluginMetaData.name}`, 400);
-  }
-  if (semver.satisfies(serverFrameworkVersion, pluginMetaData.framework, {
-    includePrerelease: true
-  })) {
-    return null;
-  }
-  return new ImportError(`Plugin ${pluginMetaData.name} (${pluginMetaData.framework}) is not compatible with the installed version (${serverFrameworkVersion})`, 400);
+/**
+ * Gets category to describe plugin framework version against installed framework version
+ * @param serverFrameworkVersion
+ * @param pluginMetaData
+ * @returns {*}
+ */
+function getPluginFrameworkVersionCategory(serverFrameworkVersion, pluginMetaData, type, callback) {
+  var pluginFrameworkVersion = pluginMetaData.framework;
+  var contentPlugin;
+
+  async.waterfall([
+      function getContentPlugin(cb) {
+          origin.contentmanager.getContentPlugin(type, cb);
+      },
+      function getDB(plugin, cb) {
+          contentPlugin = plugin;
+          database.getDatabase(cb);
+      },
+      function retrievePluginDoc(db, cb) {
+          db.retrieve(contentPlugin.bowerConfig.type, { name: pluginMetaData.name }, { jsonOnly: true }, cb);
+      },
+      function categorisePlugin(records, cb) {
+          if (records.length === 0) {
+              // Not already installed so could be:
+              // Red - this plugin is not compatible
+              // Green - this plugin is compatible and will be installed
+              if (semver.satisfies(serverFrameworkVersion, pluginFrameworkVersion, {
+                      includePrerelease: true
+                  })) {
+                return cb(null, {category: 'green', newVersion: pluginMetaData.version});
+              }
+              cb(null, {category: 'red', newVersion: app.polyglot.t('app.na')});
+          } else {
+              // A version is already installed so could be:
+              // White - same version installed as in course
+              // Amber - newer version installed and will be used in the course instead
+              // Green - older version installed so this newer version will be upgraded in the AT
+              var serverPlugin = records[0];
+              var serverPluginVersion = semver.clean(serverPlugin.version);
```

Comment:
While I remember, we should short circuit if someone has really messed up their versioning. We could probably use the red category for this.

```js
var serverPluginVersion = semver.valid(serverPlugin.version);
var importPluginVersion = semver.valid(pluginMetadata.version);

if (!serverPluginVersion || !importPluginVersion) {
  // return early
}
```